d2/src/main/java/com/linkedin/d2/balancer/clients/TrackerClientImpl.java

```diff
@@ -192,11 +192,11 @@ public void onResponse(TransportResponse<RestResponse> response)
       if (response.hasError())
       {
         Throwable throwable = response.getError();
-        handleError(_callCompletion, throwable);
+        handleError(_callCompletion, throwable, response.getWireAttributes());
```

Comment:
I think we don't need to parse server load from wireAttributes when `enableServerReportedLoad` is set to false.

Reply:
There are 3 places that we can potentially control the behavior on the client side:
1. TrackerClient - whether to pass wire attributes
2. CallTracker - whether to update the server-reported load in each interval
3. load balancer strategy - whether to take the server-reported load into load balancing decision.

When I designed this I also debated where I should put the control, in the end I only put one control in load balancer strategy to make the control logic all in one place.

Now TrackerClient and CallTracker just update the server reported load whenever it's ready, and in the load balancing strategy there are 2 cases we need to watch out:
1. Server enables reporting, but client did not enable, which we respect the client side config flag
2. Server does not enable yet, but client enables, which means the reported load will always be -1. Either case, we control the RLB to not consider server reported load.

There is definitely a way to add control in TrackerClient, where we can always report -1 as the score. There will be more code we need to cleanup later if we add a control here. What do you think?

Reply:
I just changed this logic in my latest commit. I guarded the code with the config on both TrackerClient and the hash ring selector logic.